core/src/main/java/io/questdb/griffin/SqlCompiler.java

```diff
@@ -1440,6 +1443,32 @@ private CompiledQuery createTable(final ExecutionModel model, SqlExecutionContex
         }
     }
 
+    private void copyTableReaderMetadataToCreateTableModel(SqlExecutionContext executionContext, CreateTableModel model) throws SqlException {
+        CharSequence likeTableName = model.getLikeTableName().token;
+        tableExistsOrFail(model.getLikeTableName().position, likeTableName, executionContext);
+        try (TableReader rdr = engine.getReader(executionContext.getCairoSecurityContext(), likeTableName)) {
+            model.setCommitLag(rdr.getCommitLag());
+            model.setMaxUncommittedRows(rdr.getMaxUncommittedRows());
+            TableReaderMetadata rdrMetadata = rdr.getMetadata();
+            for (int i = 0; i < rdrMetadata.getColumnCount(); i++) {
+                int symbolCapacity = rdr.getSymbolMapReader(i) != null ? rdr.getSymbolMapReader(i).getSymbolCapacity() : configuration.getDefaultSymbolCapacity();
```

Comment:
Instead of relying on `rdr.getSymbolMapReader(i) != null`, you should check the column type: `ColumnType.isSymbol(rdrMetadata.getColumnType(i))`.